Shifty.App/Components/Voucher.razor

```diff
@@ -0,0 +1,138 @@
+                                RequiredError="Product is required" 
+                                Min="1"
+                                Max="10" />
+
+                <MudTextField T="string"
+                    @bind-Value="_voucherForm.Description"
+                    Label="Description" 
+                    Required="true"
+                    RequiredError="Description is required" />
+                
+                <MudCardActions>
+                    <MudButton  Variant="Variant.Filled"
+                                Color="Color.Primary"
+                                Class="ml-auto"
+                                Disabled="@(!_isFormValid)"
+                                OnClick="@(async () => await IssueVoucher())"
+                                EndIcon="@Icons.Material.Filled.Sell">
+                        Issue Voucher
+                    </MudButton>
+                </MudCardActions>
+            </MudForm>
+            @if (_showProgressBar)
+            {
+                <MudContainer class="d-flex justify-content-center">
+                    <MudProgressCircular hidden="@_showProgressBar" Color="Color.Default" Indeterminate="true" />
+                </MudContainer>
+            }
+            @if (_showResult)
+            {
+                <MudTextField Text="@_voucherCodes"  
+                            @ref="_multilineReference"
+                            T="string"
+                            Adornment="Adornment.End" 
+                            Style="border-width: 2px; padding: 4px;" 
+                            Outlined="true" 
+                            AdornmentIcon="@Icons.Material.Outlined.ContentCopy"
+                            OnAdornmentClick="@(async () => await CopyToClipboard())" 
+                            Lines="@_vouchers.Count" 
+                            ReadOnly=true />
+            }
+        </MudCardContent>
+    </MudCard>
+</MudContainer>
+
+@code
+{
+    private VoucherForm _voucherForm = new();
+    private bool _isFormValid = false;
+    private bool _productsLoaded = false;
+    private bool _showResult = false;
+    private bool _showProgressBar = false;
+    private ICollection<ProductDto> _products = new List<ProductDto>();
+    private ICollection<IssueVoucherResponse> _vouchers;
+    private string _voucherCodes;
+    private MudTextField<string> _multilineReference;
+    
+    private class VoucherForm
+    {
+        [Required]
+        public string Description { get; set; }
+        public ProductDto Product { get; set; }
+        public int Amount { get; set; } = 1;
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        Snackbar.Configuration.PositionClass = Defaults.Classes.Position.BottomRight;
+        (_productsLoaded, _products) = await _productService.GetProducts();
+
+        if (!_productsLoaded)
+        {
+            Snackbar.Add("Fatal error loading products", Severity.Error);
```

Comment:
You could parse the error message that is returned by the API here. It should be filtered, and somewhat sensible to the end user what is returned

Reply:
Using Eithers, what is returned is a null Exception.

With the TryAsync, an error message that can happen is for instance this for status code 400 (Bad Request), when having a prefix that is not length == 3:

```
Shifty.Api.Generated.AnalogCoreV2.ApiException`1[Shifty.Api.Generated.AnalogCoreV2.ApiError]: Bad Request. See explanation
``` 

An empty "response" is then shown. So what I can filter is on the Status Code, but for now the solution shows instead a Snackbar with the message "Bad Request. See explanation Status: 400 Response:".